src/checkpoint.ts

```diff
@@ -242,13 +236,21 @@ export default class Checkpoint {
     return this.cpBlocksCache.shift();
   }
 
-  private async handleBlock(block: GetBlockResponse) {
+  private async handleBlock(block: api.RPC.GetBlockWithTxs) {
+    if (!('block_number' in block)) return;
+
     this.log.info({ blockNumber: block.block_number }, 'handling block');
 
-    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-    // @ts-ignore
-    for (const receipt of block.transaction_receipts || []) {
-      await this.handleTx(block, block.transactions[receipt.transaction_index], receipt);
+    const receipts = await Promise.all(
+      block.transactions.map(tx => {
+        if (!tx.transaction_hash) return null;
+
+        return this.provider.getTransactionReceipt(tx.transaction_hash);
+      })
+    );
```

Comment:
Additional requests required when using RPC.

Reply:
This might result in a big slowdown to sync, isn't there a way to get all events from a block within a single request?

Reply:
Maybe with `starknet_getEvents` 🤔

Reply:
@Sekhmet Yes
![image](https://user-images.githubusercontent.com/16245250/201398413-3ee7fb2c-5d0d-4643-a824-aa5e989eb552.png)

Reply:
That works, but we now miss receipt as those can only be fetched by those one-per-tx requests in RPC client so writers no longer receive them. Is that okay?

Reply:
@Sekhmet Yes I guess we can drop this feature for now. Does the event return the tx hash?

Reply:
TX hash is in both events response and blocks response.